request.js

```diff
@@ -1413,6 +1413,28 @@ Request.prototype.jar = function (jar) {
   return self
 }
 
+Request.prototype.cloneJar = function (cookieJar) {
+  var self = this
+  var newJar, cookieString, cookieStore, domains, domainPaths, cookieUrl
+
+  newJar = self.jar()
+  cookieStore = cookieJar._jar.store.idx
+  domains = Object.keys(cookieStore)
+
+  domains.forEach(function (domain) {
+    domainPaths = Object.keys(cookieStore[domain])
+
+    domainPaths.forEach(function (domainPath) {
+      cookieUrl = 'http://' + domain + domainPath
```

Comment:
I believe the protocol does matter. I would take a look at https://github.com/goinstant/tough-cookie . Seems you can set a cookie url with https. Also did you consider adding this to the `lib/cookies.js`

https://github.com/request/request/blob/a2c9ec7e28567bd58837b106abe1b9a03d881e1d/lib/cookies.js